### ISSUE TITLE
Infinite fluid disk gets overflowed.

### ISSUE THREAD

**Issue by MPThLee** [MelanX/ExtraDisks#36]
### Minecraft version

1.16.5

### Extra Disks version

1.16.4-1.5.1

### Forge version

36.2.22

### The latest.log file

(no related log found since i'm playing on server.) 

### Issue description

![javaw_2022-02-11_08-09-29](https://user-images.githubusercontent.com/26970092/153513409-91937c84-1c15-4318-972f-1b1227475fc5.png)

I just moved from Blackhole Tank from Industrial foregoing(which can hold a fluid up to [max value of 32bit signed integer]mB) after I made Infinite Fluid Disk and got this.

### Steps to reproduce

Use Sink(from Cooking form blockheads) and Ultimate Importer(from Cable Tiers) to reproduce it faster. 

### Other information

I'm currently playing on All the mods 6 pack with version 1.8.17

If using disk manipulator with overflowed disk, It duplicates disk(but same id) and sometime it loses a data.

Also overflowed disk gets "red" signal on disk drive. 

Here's contents of disk:
![image](https://user-images.githubusercontent.com/26970092/153513919-55cab997-d292-4206-afa3-42df12c844c6.png)
Water goes `0` after it actually overflowed.
`Honey` from `Resourceful bees` almost near the max of 32 bit int.

**Comment by James103** [comment]
Can reproduce in Project Architect 1.7.

To reproduce:
1. Use the listed modpack. The modpack can be found at https://www.curseforge.com/minecraft/modpacks/project-architect
2. Create a new world.
3. Set up a Refined Storage system with a controller, a power source, a Fluid Grid, and an Infinite Fluid Storage Block.
4. Set up a creative tank full of lava. For fastest reproduction, use many Entangled Blocks attached to the creative tank and attach Importers (each with max speed upgrades) to each Entangled Block.
5. Wait...for several real-life hours.
6. Notice that all lava stored in the disk is deleted once the amount of lava stored hits 2,147,483,647 mB.

The mod list is part of a dummy crash report which is attached below:
[example_crash.txt](https://github.com/MelanX/ExtraDisks/files/8191628/example_crash.txt)

Possible fix:
Cap the amount of a single fluid in an Infinite Storage Disk or Block to 2,147,483,647 mB.
In other words, the Infinite Storage disks should be able to hold an infinite number of item or fluid types, but a maximum of 2,147,483,647 for each item or fluid type.

**Comment by MelanX** [comment]
I created a pull request for this issue on Refined Storage as you can see in the timeline here. Once it's merged, you just need to update to fix this problem.
Sadly, this is only for 1.18.2. Since you're using 1.16.5 modpacks, you need to hope that this will also be fixed in the 1.16.5 version of RS.

**Comment by Eq1s** [comment]
same thing, but with item disk
after there are more items of the same type than the int value, it goes into the negative side and the items are sold, and then it can go from a negative value to a positive one (also if a constant flow of items goes to the network, then every time you re-enter the terminal, the amount of the item will be reset to zero and accumulate while you are in the terminal interface)
![2022-05-30_15 16 04](https://user-images.githubusercontent.com/95995139/170974620-84e82d77-6e23-427f-8c50-0833aa9fb183.png)
~~atm 6 - to the sky, but im think it doesn't matter~~

**Comment by MelanX** [comment]
Fixed in next RS version: https://github.com/refinedmods/refinedstorage/commit/c66a17452c669228da6333f44e3af268e0163492
